packages/wdio-browserstack-service/src/reporter.ts

```diff
@@ -27,34 +29,69 @@ export default class TestReporter extends WDIOReporter {
+
+        switch (this._config?.framework) {
+        case 'mocha':
+            return event === 'skip'
+        case 'cucumber':
+            return false
+        case 'jasmine':
+            return testType === 'test'
+        default:
+            return false
+        }
+    }
+
+    async onTestEnd(testStats: TestStats) {
+        if (!this.needToSendData('test', 'end')) return
+
+        testStats.end ||= new Date()
+        await this.sendTestRunEvent(testStats, 'TestRunFinished')
+    }
+
+    async onTestStart(testStats: TestStats) {
+        if (!this.needToSendData('test', 'start')) return
+
+        this._tests[testStats.fullTitle] = {
+            uuid: uuidv4(),
+        }
+        await this.sendTestRunEvent(testStats, 'TestRunStarted')
+    }
+
+    async sendTestRunEvent(testStats: TestStats, eventType: string) {
         const framework = this._config?.framework
 
-        if (this._observability && framework !== 'cucumber') {
-            let testData: TestData = {
-                uuid: uuidv4(),
-                type: testStats.type,
-                name: testStats.title,
-                body: {
-                    lang: 'webdriverio',
-                    code: null
-                },
-                scope: testStats.fullTitle,
-                scopes: getHierarchy(testStats.fullTitle),
-                identifier: testStats.fullTitle,
-                file_name: this._suiteName ? path.relative(process.cwd(), this._suiteName) : undefined,
-                location: this._suiteName ? path.relative(process.cwd(), this._suiteName) : undefined,
-                started_at: (new Date()).toISOString(),
-                framework: framework,
-                finished_at: (new Date()).toISOString(),
-                duration_in_ms: testStats._duration,
-                retries: { limit:0, attempts: 0 },
-                result: testStats.state,
-            }
+        let testData: TestData = {
+            uuid: this._tests[testStats.fullTitle].uuid || uuidv4(),
+            type: testStats.type,
+            name: testStats.title,
+            body: {
+                lang: 'webdriverio',
+                code: null
+            },
+            scope: testStats.fullTitle,
+            scopes: this._suites.map(s => s.title),
+            identifier: testStats.fullTitle,
+            file_name: this._suiteName ? path.relative(process.cwd(), this._suiteName) : undefined,
+            location: this._suiteName ? path.relative(process.cwd(), this._suiteName) : undefined,
+            started_at: testStats.start && testStats.start.toISOString(),
+            finished_at: testStats.end && testStats.end.toISOString(),
+            framework: framework,
+            duration_in_ms: testStats._duration,
+            result: testStats.state,
+            retries: { limit: testStats.retries || 0, attempts: testStats.retries || 0 }
+        }
 
+        if (eventType == 'TestRunStarted' || eventType == 'TestRunSkipped') {
```

Comment:
Why is this check specifically for these 2 events?

Reply:
We are adding the integrations object here. It is only required for new tests right? That's what we used to do in insight_handler before also.

Reply:
Let's add this for TestRunFinished event also in case a user spawns driver inside the test itself in which case currently we would not be sending any integrations data.

Reply:
Added